### PR TITLE
solution

### PR DESCRIPTION
**Pull request by VladBrynzia**
[DEMO LINK](https://vladbrynzia.github.io/react_add-todo-form/)

### CODE REVIEW

**Review by Olexandr-Trubin-Koshyk — APPROVED** [review]
Well done

File: src/App.tsx [hunk 1]
```diff
@@ -1,18 +1,87 @@
...
+import todosFromServer from './api/todos';
 import users from './api/users';
+import { TodoList } from './components/TodoLits';
 
 const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userIdError, setUserIdError] = useState(false);
+  const copyTodosFromServer = [...todosFromServer];
```
Comment (Ihor-Karpyn) [inline src/App.tsx:12]:
```suggestion
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,18 +1,87 @@
...
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
+    <>
+      <form onSubmit={onChange}>
```
Comment (Ihor-Karpyn) [inline src/App.tsx:42]: why do you use `onChange` function on form submit?

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -1,18 +1,86 @@
...
+  const [userIdError, setUserIdError] = useState(false);
+
+  const [todos, setTodos] = useState(todosFromServer);
+
+  const onSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (userId && title) {
+      setTodos((currentState) => {
+        const createTodo = {
```
Comment (solaryasha) [inline src/App.tsx:20]:
```suggestion
        const createdTodo = {
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,18 +1,86 @@
...
+  const [titleError, setTitleError] = useState(false);
+  const [userIdError, setUserIdError] = useState(false);
+
+  const [todos, setTodos] = useState(todosFromServer);
+
+  const onSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (userId && title) {
+      setTodos((currentState) => {
```
Comment (solaryasha) [inline src/App.tsx:19]:
```suggestion
      setTodos((previousTodos) => {
```

File: src/App.tsx [hunk 5]
```diff
@@ -1,18 +1,86 @@
...
-        {users.length}
-      </p>
-    </div>
+    <>
+      <form onSubmit={onSubmit}>
+        <select
+          name="select"
+          value={userId}
+          onChange={event => {
+            setUserId(+event.target.value);
```
Comment (solaryasha) [inline src/App.tsx:46]:
```suggestion
            setUserId(Number(event.target.value));
```

File: src/components/TodoInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo
+};
+
+export const TodoInfo: React.FC<Props> = ({ todo }) => {
```
Comment (solaryasha) [inline src/components/TodoInfo.tsx:8]: use react.memo

File: src/components/TodoLits.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { TodoInfo } from './TodoInfo';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todos: Todo[]
+};
+
+export const TodoList: React.FC<Props> = ({ todos }) => {
```
Comment (solaryasha) [inline src/components/TodoLits.tsx:9]: react.memo

**Review by solaryasha — APPROVED** [review]

**Pull request closed without merging**